pkg/scalers/azure_servicebus_scaler.go

```diff
@@ -233,105 +230,96 @@ func (s *azureServiceBusScaler) GetMetrics(ctx context.Context, metricName strin
 	return append([]external_metrics.ExternalMetricValue{}, metric), nil
 }
 
-type azureTokenProvider struct {
-	httpClient  *http.Client
-	ctx         context.Context
-	podIdentity kedav1alpha1.AuthPodIdentity
-}
-
-// GetToken implements TokenProvider interface for azureTokenProvider
-func (a azureTokenProvider) GetToken(uri string) (*auth.Token, error) {
-	ctx := a.ctx
-
-	var token azure.AADToken
-	var err error
-
-	switch a.podIdentity.Provider {
-	case kedav1alpha1.PodIdentityProviderAzure:
-		token, err = azure.GetAzureADPodIdentityToken(ctx, a.httpClient, a.podIdentity.IdentityID, serviceBusResource)
-	case kedav1alpha1.PodIdentityProviderAzureWorkload:
-		token, err = azure.GetAzureADWorkloadIdentityToken(ctx, a.podIdentity.IdentityID, serviceBusResource)
-	default:
-		err = fmt.Errorf("unknown pod identity provider")
-	}
-	if err != nil {
-		return nil, err
-	}
-
-	return auth.NewToken(auth.CBSTokenTypeJWT, token.AccessToken, token.ExpiresOn), nil
-}
-
 // Returns the length of the queue or subscription
 func (s *azureServiceBusScaler) getAzureServiceBusLength(ctx context.Context) (int64, error) {
-	// get namespace
-	namespace, err := s.getServiceBusNamespace(ctx)
+	// get adminClient
+	adminClient, err := s.getServiceBusAdminClient(ctx)
 	if err != nil {
 		return -1, err
 	}
 	// switch case for queue vs topic here
 	switch s.metadata.entityType {
 	case queue:
-		return getQueueEntityFromNamespace(ctx, namespace, s.metadata.queueName)
+		return getQueueLength(ctx, adminClient, s.metadata.queueName)
 	case subscription:
-		return getSubscriptionEntityFromNamespace(ctx, namespace, s.metadata.topicName, s.metadata.subscriptionName)
+		return getSubscriptionLength(ctx, adminClient, s.metadata.topicName, s.metadata.subscriptionName)
 	default:
 		return -1, fmt.Errorf("no entity type")
 	}
 }
 
 // Returns service bus namespace object
-func (s *azureServiceBusScaler) getServiceBusNamespace(ctx context.Context) (*servicebus.Namespace, error) {
-	var namespace *servicebus.Namespace
+func (s *azureServiceBusScaler) getServiceBusAdminClient(ctx context.Context) (*admin.Client, error) {
+	if s.client != nil {
+		return s.client, nil
+	}
+
+	var adminClient *admin.Client
 	var err error
 
 	switch s.podIdentity.Provider {
 	case "", kedav1alpha1.PodIdentityProviderNone:
-		namespace, err = servicebus.NewNamespace(servicebus.NamespaceWithConnectionString(s.metadata.connection))
+		adminClient, err = admin.NewClientFromConnectionString(s.metadata.connection, nil)
 		if err != nil {
-			return namespace, err
+			return nil, err
 		}
-	case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
-		namespace, err = servicebus.NewNamespace()
+	case kedav1alpha1.PodIdentityProviderAzure:
+		// Once azure-sdk-for-go supports Workload Identity we can use this for Workload identity too
+		chain, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
Could you also pass in `s.podIdentity.identityID` into using `ManagedCredentials` here for the `Provider: azure` case? The user might want to use an identity other than the one they configured KEDA with.

Reply:
hum... ATM they don't support it with default credentials. I have to create another `NewChainedTokenCredential` mixing that with the CLI like I did with wi because they only supports getting the clientId from environment.  
This is really annoying :/

Reply:
TBH, I'm thinking in mixing them right now into a single case with the 3 token providers. I have to add them one by one to support providing the ID so this could be easier

Reply:
I'll do it. Having more than 1 `TokenCredential` is not a problem because it caches the first one which has enough access, so we can simplify the code creating a single chained credential

Reply:
Tbh, this is my fault. Using the single chained credential that you did in the first place was the simpler way. Sorry.